libs/feature/teaching/src/lib/skills/folder-editor/cycle-detection/detection-hook.tsx

```diff
@@ -0,0 +1,68 @@
+import {FolderItem} from "./cycle-detection";
+import {useEffect, useState} from "react";
+
+
+
+interface DrawState {
+	item: FolderItem | null,
+	initial: boolean
+}
+
+const listener: Map<string, Array<(state: DrawState) => void>> = new Map();
+const memorizedDetection: FolderItem[] = [];
+
+export const dispatchDetection = (items: FolderItem[]) => {
+	items.forEach(item => {
+		const listenerCallbacks = listener.get(item.skill.id);
+		if (listenerCallbacks) {
+			if(item.cycle || item.parent) {
+				memorizedDetection.push(item);
+			}
+			for(const callback of listenerCallbacks) {
+				callback({item: item, initial: false});
+			}
+		}
+	});
+};
+
+export const ClearCycleDetection = () => {
```

Comment:
methodenname sollte in lower camel case geschrieben werden, sofern es sich nicht um eine react komponente handelt.